app/models/qdm/patient.rb

```diff
@@ -0,0 +1,143 @@
+class Qdm::Patient
+  include Mongoid::Document
+  field :birth_datetime, type: DateTime
+  field :qdm_version, type: String, default: '5.3'
+  field :ethnicity, type: String
+  field :race, type: String
+  field :sex, type: String
+
+  # These are the "data criteria", or QDM datatype elements that exist on a
+  # patient.
+  embeds_many :history_elements
+
+  # Returns an array of history elements that exist on this patient, that
+  # match the given OID.
+  def get_by_oid(oid)
+    history_elements.where(oid: oid) || []
+  end
+
+  # Returns an array of history elements that exist on this patient. Optionally
+  # takes a type, which returns all history elements of that QDM category.
+  # Example: patient.get_history_elements(type='encounters') will return all
+  # Encounter QDM data types active on the patient.
+  def get_history_elements(type=nil)
+    if type
+      history_elements.where(category: type)
```

Comment:
rather than do a `|| []` in each getter, does it make more sense to guarantee that `get_history_elements` will always return an `Array`? That way all the `history_element` getters will be consistent too.

Reply:
That does make good sense, I'll put in that change.